### PR TITLE
Fix erc20 chain deposits

### PR DESCRIPTION
**Pull request by kerzhner**
Fixes https://github.com/statechannels/go-nitro/issues/755

This PR modifies the EthChainService to only inlclude ETH with a native token deposit. The logic change is [fairly small](https://github.com/statechannels/go-nitro/blob/95df1f8eb13350e9f4a2920ac59abf88fb80154d/client/engine/chainservice/eth_chainservice.go#L65). The bulk of the PR is:
- erc20 contract bindings are added to `go-nitro`
- the new bindings are used to test the logic change.
- [refactor of DepositedEvent](https://github.com/statechannels/go-nitro/pull/763/commits/1424f2409898a238a1c56538215b3c7bf64ff010)

### CODE REVIEW

File: client/engine/chainservice/eth_chainservice.go [hunk 1]
```diff
@@ -44,24 +44,32 @@ func NewEthChainService(chain ethChain, na *NitroAdjudicator.NitroAdjudicator, n
...
+}
+
+// SendTransaction sends the transaction and blocks until it has been submitted.
+func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) *ethTypes.Transaction {
 	switch tx := tx.(type) {
 	case protocols.DepositTransaction:
 		for tokenAddress, amount := range tx.Deposit {
-			txOpt.Value = amount
+			txOpts := ecs.defaultTxOpts()
+			ethTokenAddress := common.Address{}
```
Comment (geoknee) [inline client/engine/chainservice/eth_chainservice.go:64]: Should this be extracted somewhere more central? We rely on it in many places...

File: client/engine/chainservice/eth_chainservice.go [hunk 2]
```diff
@@ -75,7 +83,8 @@ func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) *ethT
 		nitroVariablePart := NitroAdjudicator.ConvertVariablePart(state.VariablePart())
 		nitroSignatures := []NitroAdjudicator.IForceMoveSignature{NitroAdjudicator.ConvertSignature(signatures[0]), NitroAdjudicator.ConvertSignature(signatures[1])}
 
-		ethTx, err := ecs.na.ConcludeAndTransferAllAssets(&txOpt, nitroFixedPart, nitroVariablePart, 1, []uint8{0, 0}, nitroSignatures)
+		txOpts := ecs.defaultTxOpts()
```
Comment (geoknee) [inline client/engine/chainservice/eth_chainservice.go:87]: You could return a `bind.TransactOpts` from defaultTxOpts and then you could inline this function call into L87? Just an idea.

File: client/engine/chainservice/mockchain_test.go [hunk 3]
```diff
@@ -55,31 +50,24 @@ func TestDeposit(t *testing.T) {
...
 	eventB = <-eventFeedB
+	receivedEventIsValid(t, eventB, expectedHoldings, testTx.ChannelId())
+}
 
-	if eventB.ChannelID() != testTx.ChannelId() {
-		t.Fatalf(`channelId mismatch: expected %v but got %v`, testTx.ChannelId(), eventB.ChannelID())
-	}
-	if !eventB.(DepositedEvent).Holdings.Equal(expectedHoldings) {
-		t.Fatalf(`holdings mismatch: expected %v but got %v`, expectedHoldings, eventB.(DepositedEvent).Holdings)
+func receivedEventIsValid(t *testing.T, receivedEvent Event, holdings types.Funds, channelId types.Destination) {
```
Comment (geoknee) [inline client/engine/chainservice/mockchain_test.go:64]: Maybe `checkReceivedEventIsValid`?

**Review by geoknee — APPROVED** [review]
Should the chainservice be responsible for doing the erc20 `approve` transaction? Currently, that is done manually in the test script.

File: client/engine/chainservice/eth_chainservice.go (hunk 1 above)
Comment (kerzhner) [reply client/engine/chainservice/eth_chainservice.go:64]: Good call, filed https://github.com/statechannels/go-nitro/issues/766.

File: client/engine/chainservice/eth_chainservice.go (hunk 2 above)
Comment (kerzhner) [reply client/engine/chainservice/eth_chainservice.go]: Good call! 8436395c

File: client/engine/chainservice/mockchain_test.go (hunk 3 above)
Comment (kerzhner) [reply client/engine/chainservice/mockchain_test.go]: Done in a4274fa3

**Comment by kerzhner** [comment]
> Should the chainservice be responsible for doing the erc20 `approve` transaction? Currently, that is done manually in the test script.

You are right, we should be approving erc20 tokens in the chainservice. Let's do that in a separate PR as the changeset will be sizeable.

**Comment by kerzhner** [comment]
ERC20 token approval comment is addressed in https://github.com/statechannels/go-nitro/pull/767

**Pull request merged**